### ISSUE TITLE
Could I use this model to solve the 3-D image classification problem?

### ISSUE THREAD

**Issue by lisen0** [ellisdg/3DUnetCNN#34]

**Comment by ellisdg** [comment]
I don't know what specific problem you are talking about, but this model is used for image segmentation. Which means it can be trained to locate brain tumors in an image, classify tissue types, etc. That means this model takes in a 3D input and then outputs a 3D image of binary labels. If you want to take in a 3D image and output 1D binary labels, then this is not the model you want.

**Comment by lisen0** [comment]
What I want to solve is, given a 3-D image of retina, I need to determine if it is normal. I need the model to output 0 or 1. So could I make any changes to your model to satisfy my need? Like adding softmax layer?

**Comment by lisen0** [comment]
Could you please give me some advice about that? I am doing 3-D medical image classification now. But I don't know which approach I should choose. Can I use timedistributed+conv2d to solve that problem?

**Comment by ellisdg** [comment]
I don't know what is going to work best for your application, as I've never tried to do something like that, but I can try to help. Check out the shape of the [UNet](https://github.com/ellisdg/3DUnetCNN/blob/master/doc/3dUNet.png). Just past the bottom of the U-shape, it starts to up-sample (or 'Deconvolution', which is sort of like a reverse convolution) in order to produce an image that is the same shape as the input. Since you just want a 0 or 1 for each image, you could cut the UNet off before the up-sampling and then add some convolution or fully connected layers.

You should check out the neural network architecture listed here:
https://github.com/dhammack/DSB2017/blob/master/dsb_2017_daniel_hammack.pdf

I think that would be pretty close to what you want.

**Comment by lisen0** [comment]
Thank you very much for your reply. This will help me a lot.

**Comment by divinediu** [comment]
@lisen0 What did you use for your classification problem?
